pyproject.toml

```diff
@@ -78,6 +78,7 @@ spacy = {optional = true, version = "^3.7.1"}
 aiohttp = "^3.8.6"
 networkx = ">=3.0"
 psycopg2-binary = {optional = true, version = "^2.9.9"}
+dirtyjson = "^1.0.8"
```

Comment:
We still do. Lazy import is still import, not installation. We have to install it before we can import it.

Reply:
could we move it to `[tool.poetry.group.dev.dependencies]` instead of `[tool.poetry.dependencies]`

Reply:
No, because it is a run-time dependency, not a development-time-only dependency.

Reply:
understand. we try to avoid adding many deps for core pkgs. this dirtyjson pkg should be only used when users need to call this react output parser (we don't have other places using it). so I would suggest move it to dev-time dep so that you could pass unit tests with this pkg.

Reply:
cc @logan-markewich if you have better suggestion

Reply:
That's unfortunately not how `tool.poetry.group.dev.dependencies` works.

Reply:
Do we actually need this package though? It would be nice if we could accomplish something similar without it. Plus, that package seems pretty unmaintained 🤔

Reply:
I think there's a few places that could use improved json parsing. If there's something more popular/maintained, I would be more open to adding that 👀

Reply:
Spent a while looking for alternatives. Just judging from their READMEs:
* `jsonrepair`: closest to what I need, but it’s only in TypeScript.
* `demjson`: even less maintained. The homepage doesn’t even load for me.
* `GSON`: Java only.
* `jsonic`: also abandoned for ~2 years. Worse, it’s not meant for recovering JSON objects.
* `json5`: it’s another standard, not meant for recovering JSON objects.

So I’d still say `dirtyjson` is our best bet here.

> Do we actually need this package though? 

Quite important for making those “7B” open source LLMs work practically, TBH. I can testify.

> It would be nice if we could accomplish something similar without it.

Hmmm. I’m not a fan of spending too much effort on solving a problem that is already solved.

Honestly, `dirtyjson` being abandoned doesn’t really bother me (though I’m usually that kind of person who enjoys `apt upgrade`-ing all my boxes frequently), because this package solves one particular **timeless** problem quite well. What I mean is that, all the quirky JSON mistakes people would practically made, most likely have already been made in the past decade, so the “ruleset” in `dirtyjson` should be quite tried-and-true. It is not like it that in 2024 someone’s gonna fail at writing a valid JSON in a spectacularly innovative way (or with any statistical significance).

I think the fact that `dirtyjson` hasn’t been updated for a couple of years actually is a proof that it is mature.

Reply:
true true. 

Ok, I think I'm sold. My one caveat is that I (or someone else) needs to update some of our other json-parsing functions to use this

Reply:
Thanks @tslmy !